decentraland/screens/BuyLandScreen.js

```diff
@@ -29,55 +33,52 @@ export class BuyLandScreen extends React.Component {
   };
 
   _buy = landForSale => {
-    const { navigation, account, removeLandForSale } = this.props;
+    const { props, _executeOrder } = this;
+    const { navigation, account, removeLandForSale } = props;
     const onSuccess = () => {
       removeLandForSale(landForSale);
     };
-    this._executeOrder(landForSale, account, onSuccess);
+
+    _executeOrder(landForSale, account, onSuccess);
     navigation.navigate("ListLandForSaleScreen");
   };
 
   _executeOrder = async (sellOrder, account, afterSuccessfulExecution) => {
-    const { asset } = sellOrder;
-    const { priceMana } = sellOrder;
-    const priceInWei = Number(priceMana) * 1e18;
-
-    const ONE = 1e18;
-    const TEN = 10e18;
-
-    // TODO: Go deep on gas handling.
-    // Without that, VM returns a revert error instead of out of gas error.
-    // See: https://github.com/tasitlabs/TasitSDK/issues/173
-    const gasParams = {
-      gasLimit: 7e6,
-      gasPrice: 1e9,
-    };
-
-    const marketplaceAddress = this.marketplaceContract.getAddress();
-    const estateAddress = this.estateContract.getAddress();
-
-    await manaFaucetTo(account, TEN);
-
-    // TODO: Move this out to an explicit approval flow?
-    await approveManaSpending(account, marketplaceAddress, ONE);
-
-    this.marketplaceContract.setWallet(account);
-
-    const fingerprint = await this.estateContract.getFingerprint(asset.id);
-
-    const executeOrder = this.marketplaceContract.safeExecuteOrder(
-      estateAddress,
-      asset.id,
-      priceInWei.toString(),
-      fingerprint.toString(),
-      gasParams
-    );
-
-    await executeOrder.waitForNonceToUpdate();
-
-    // TODO: This function should be called inside of the eventListener
-    // that catches the safeExecuteOrder successful event.
-    afterSuccessfulExecution();
+    try {
+      const { priceMana, asset, type } = sellOrder;
+      const { id: assetId } = asset;
+      const priceInWei = Number(priceMana) * 1e18;
+      const contracts = getContracts();
+      const { marketplaceContract, estateContract, landContract } = contracts;
+
+      let nftAddress;
+      let fingerprint;
+      if (type == ESTATE) {
+        nftAddress = estateContract.getAddress();
+        fingerprint = await estateContract.getFingerprint(assetId);
+      } else if (type == PARCEL) {
+        nftAddress = landContract.getAddress();
+        // LANDRegistry contract doesn't implement getFingerprint function
+        fingerprint = "0x";
+      }
+
+      marketplaceContract.setWallet(account);
+      const action = marketplaceContract.safeExecuteOrder(
+        nftAddress,
+        `${assetId}`,
+        `${priceInWei}`,
+        `${fingerprint}`,
+        gasParams
+      );
+
+      // TODO: This function should be called inside of the eventListener
+      // that catches the safeExecuteOrder successful event.
+      await action.waitForNonceToUpdate();
+      showInfo(`Asset bought successfully.`);
```

Comment:
Maybe dynamically show land or estate based on what was bought?

Reply:
I'm not sure if that is what you've meant but after the purchase, the asset is being hidden from the main list.

Reply:
It's good to know that the asset is being hidden from the main list.

What I mean is more like the following in pseudocode

```
if asset type is land
      showInfo(`Land bought successfully.`);
else if asset type is estate
      showInfo(`Estate bought successfully.`);
```

Reply:
I'll do that

Reply:
Done on PR https://github.com/tasitlabs/tasit/pull/161

Reply:
Thanks! Resolving